youtube_dl/extractor/common.py

```diff
@@ -826,6 +826,24 @@ def _get_netrc_login_info(self, netrc_machine=None):
             except (IOError, netrc.NetrcParseError) as err:
                 self._downloader.report_warning(
                     'parsing .netrc: %s' % error_to_compat_str(err))
+        
+        if self._downloader.params.get('usekeyring', False):
+            try:
+                import gi
+                gi.require_version('Secret', '1')
+                from gi.repository import Secret
+                LIBSECRET_SCHEMA = Secret.Schema.new("io.github.rg3.youtube-dl.Store",Secret.SchemaFlags.DONT_MATCH_NAME,{"extractor": Secret.SchemaAttributeType.STRING,"app": Secret.SchemaAttributeType.STRING})
+                secret = Secret.password_lookup_sync(LIBSECRET_SCHEMA, {"extractor": "json", "app": "youtube-dl"}, None)
+                if secret is None:
+                    raise netrc.NetrcParseError('Can\'t find credentials in keyring for "' + netrc_machine + '"')
+                print(str(secret))
+                secret_dict = json.loads(secret)
+                print('JSON Parsed: ' + str(secret_json))
+                print('JSON User: ' + secret_json['u'])
+                print('JSON Password: ' + secret_json['p'])
+            except (IOError, netrc.NetrcParseError) as err:
```

Comment:
ImportError and JSON parsing errors should be handled as well

Reply:
Added handling https://github.com/rg3/youtube-dl/commit/3be4a1db6b6e144390cb337888daba6c4b52c852